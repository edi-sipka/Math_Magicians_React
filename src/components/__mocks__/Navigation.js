/* eslint-disable */
import React from 'react';
import { BrowserRouter, NavLink } from 'react-router-dom';
import styles from '../Navigation.module.css';

const Navbar = [
  { to: '/', text: 'Home' },
  { to: '/calculator', text: 'Calculator' },
  { to: '/quote', text: 'Quote' },
];

const Nav = () => (
  <BrowserRouter>
    <nav className={styles.navigation}>
      <h1>Math Magicians</h1>
      <ul className={styles.item}>
        {Navbar.map((link) => (
          <li key={link.to}>
            <NavLink to={link.to} className={styles.links}>
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  </BrowserRouter>
);

export default Nav;
