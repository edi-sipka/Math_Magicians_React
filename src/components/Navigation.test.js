/* eslint-disable */
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navigation from './__mocks__/Navigation';

it('Navigation links are working', () => {
  const tree = renderer.create(<Navigation />).toJSON();
  expect(tree).toMatchSnapshot();
});
