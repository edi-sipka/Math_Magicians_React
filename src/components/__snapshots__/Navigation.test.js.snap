// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation links are working 1`] = `
<nav
  className="navigation"
>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="item"
  >
    <li>
      <a
        aria-current="page"
        className="links active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="links"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className="links"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
